### PR TITLE
DOC: update the CategoricalIndex docstring

### PR DESCRIPTION
**Pull request by Adrien2112**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
################################################################################
################################## Validation ##################################
################################################################################

Errors found:
        No returns section found
        No examples section found
```

The documentation has been improved. The function actually refers to a function defined elsewhere. Therefore, we decided to include a "see also" section. 

Same message for CategoricalIndex.codes, CategoricalIndex.ordered and CategoricalIndex.categories.

### CODE REVIEW

File: pandas/core/indexes/category.py [hunk 1]
```diff
@@ -32,27 +32,65 @@
...
     Parameters
     ----------
     data : array-like or Categorical, (1-dimensional)
     categories : optional, array-like
-        categories for the CategoricalIndex
+        Categories for the CategoricalIndex.
     ordered : boolean,
-        designating if the categories are ordered
+        Designating if the categories are ordered.
     copy : bool
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/category.py]: default False

File: pandas/core/indexes/category.py [hunk 2]
```diff
@@ -32,27 +32,65 @@
...
-        Make a copy of input ndarray
+        Make a copy of input ndarray.
     name : object
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/category.py]: add `, optional`

File: pandas/core/indexes/category.py [hunk 3]
```diff
@@ -32,27 +32,65 @@
...
-    categories
-    ordered
+    codes : ndarray
+        The codes (integer positions, which point to the categories) of this
+        CategoricalIndex, read only.
+    categories : Index
+        The categories of this CategoricalIndex.
+    ordered : boolean
+        Whether or not this CategoricalIndex is ordered. An ordered
+        CategoricalIndex allows to have .min() and .max() values.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/category.py]: Not something you could know (sphinx detail related to our docs), but: you don't need to include this, as the html generation will automatically turn this into a table with those explanations added.

File: pandas/core/indexes/category.py [hunk 4]
```diff
@@ -32,27 +32,65 @@
...
+
+    Examples
+    --------
+    >>> pd.CategoricalIndex(['a','b','c','a','b','c'], ordered=False,
+    ...                     categories=['c', 'b', 'a'])
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/category.py]: I would start with showing an example where you don't pass `categories` and `ordered` to explain that by default, those are inferred from the data.

Then you can show in the next examples how to overwrite those defaults with those keywords

File: pandas/core/indexes/category.py [hunk 5]
```diff
@@ -309,14 +347,44 @@ def tolist(self):
 
     @property
     def codes(self):
+        """
+        Return codes attribute of self._data.
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/category.py]: We should not refer to `self._data` as this is a private implementation detail the user should not be aware of.
 
But you can use the explanation you put above in the CategoricalIndex docstring (the part I asked you to remove again)

And similar comment for `categories` and `ordered` below

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR! Added some comments

**Comment by jorisvandenbossche** [comment]
@Adrienlq any reason you are closing this?

**Comment by Adrienlq** [comment]
Hello, sorry I have very few experience with github. I thought the PR was a failure so I closed it.
What is the status of the PR ? Should I do something now ?
Thank you

**Comment by jorisvandenbossche** [comment]
@Adrienlq The PR is certainly not a failure! It's a good start, but with some to do's. I added comments about that, and if you would like to further work on it, you can make changes based on this feedback, add this as a commit in your branch, and push this to github, which will automatically update this PR. 

See https://python-sprints.github.io/pandas/guide/pandas_pr.html#updating-a-pull-request for some more detailed information

**Comment by jreback** [comment]
closing as stale. if you'd like to continue pls ping.

**Pull request closed without merging**